api/src/data/widgetUtils.ts

```diff
@@ -7,43 +7,78 @@ import {
 } from '../db/models';
 import Messages from '../db/models/ConversationMessages';
 import { IBrowserInfo } from '../db/models/Customers';
+import { KIND_CHOICES } from '../db/models/definitions/constants';
+import { getVisitorLog } from './logUtils';
 
 export const getOrCreateEngageMessage = async (
-  customerId: string,
-  browserInfo: IBrowserInfo
+  browserInfo: IBrowserInfo,
+  visitorId?: string,
+  customerId?: string
 ) => {
-  const customer = await Customers.getCustomer(customerId);
+  let integrationId;
+
+  let customer;
+
+  if (customerId) {
+    customer = await Customers.getCustomer(customerId);
+    integrationId = customer.integrationId;
+  }
+
+  let visitor;
+
+  if (visitorId) {
+    visitor = await getVisitorLog(visitorId);
+    integrationId = visitor.integrationId;
+  }
 
   // Preventing from displaying non messenger integrations like form's messages
   // as last unread message
   const integration = await Integrations.findOne({
```

Comment:
use Integrations.getIntegration(selector) static function which checks existence of the integration inside